libcob/call.c

```diff
@@ -1290,7 +1290,7 @@ cob_cancel_field (const cob_field *f, const struct cob_call_struct *cs)
 #pragma warning(suppress: 4113) /* funcint is a generic function prototype */
 				cancel_func = s->cob_cstr_cancel.funcint;
 #else
-				cancel_func = s->cob_cstr_cancel.funcint;
+				cancel_func = (int (*)(const int,  void *, void *, void *, void *))s->cob_cstr_cancel.funcint;
 #endif
 				(void)cancel_func (-1, NULL, NULL, NULL,
 						   NULL);
```

Comment:
I have no clue why the cancel_func is setup to have 4 NULL, especially if we have a program with no USING then the cancel_func has only a single parameter `const int entry`, no?
Telling C explicit how many arguments to handle and then ask it to handle too much seems to ask for trouble...